src/main/scala/nl/knaw/dans/easy/multideposit/actions/AddPropertiesToDeposit.scala

```diff
@@ -35,66 +34,94 @@ case class AddPropertiesToDeposit(row: Int, entry: (DatasetID, Dataset))(implici
+  private def writeProperties(depositorUserID: String, emailaddress: DatamanagerEmailaddress)(implicit settings: Settings): Try[Unit] = {
     val props = new Properties {
       // Make sure we get sorted output, which is better readable than random
       override def keys(): ju.Enumeration[AnyRef] = Collections.enumeration(new ju.TreeSet[Object](super.keySet()))
     }
 
-    addProperties(props, dataset, settings.datamanager, emailaddress)
+    Try { addProperties(props, depositorUserID, settings.datamanager, emailaddress) }
       .flatMap(_ => Using.fileWriter(encoding)(stagingPropertiesFile(datasetID)).map(out => props.store(out, "")).tried)
       .recoverWith {
         case NonFatal(e) => Failure(ActionException(row, s"Could not write properties to file: $e", e))
       }
   }
 
-  def addProperties(properties: Properties, dataset: Dataset, datamanager: String, emailaddress: DatamanagerEmailaddress): Try[Unit] = {
-    for {
-      depositorUserID <- validateDepositor(dataset)
-      _ = properties.setProperty("state.label", "SUBMITTED")
-      _ = properties.setProperty("state.description", "Deposit is valid and ready for post-submission processing")
-      _ = properties.setProperty("depositor.userId", depositorUserID)
-      _ = properties.setProperty("datamanager.userId", datamanager)
-      _ = properties.setProperty("datamanager.email", emailaddress)
-    } yield ()
+  private def addProperties(properties: Properties, depositorUserID: String, datamanager: String, emailaddress: DatamanagerEmailaddress): Unit = {
+    val sf = getSpringfieldData
+    val props: Map[String, Option[String]] = Map(
+      "state.label" -> Some("SUBMITTED"),
+      "state.description" -> Some("Deposit is valid and ready for post-submission processing"),
+      "depositor.userId" -> Some(depositorUserID),
+      "datamanager.userId" -> Some(datamanager),
+      "datamanager.email" -> Some(emailaddress),
+      "springfield.domain" -> sf.map(_.domain),
+      "springfield.user" -> sf.map(_.user),
+      "springfield.collection" -> sf.map(_.collection)
+    )
+
+    for ((key, value) <- props.collect { case (k, Some(v)) => (k, v) }) {
+      properties.setProperty(key, value)
+    }
   }
 
-  private def validateDepositor(dataset: Dataset) = {
+  private def getDepositorID(dataset: Dataset): Try[String] = {
     dataset.get("DEPOSITOR_ID")
       .flatMap(_.headOption)
       .map(Success(_))
       .getOrElse(Failure(new IllegalStateException("""The column "DEPOSITOR_ID" is not present""")))
   }
+
+  /**
+   * @return Retrieve the springfield data from the dataset iff the springfield data is present
+   */
+  private def getSpringfieldData: Option[SpringfieldData] = {
+    for {
+      user <- dataset.findValue("SF_USER")
+      collection <- dataset.findValue("SF_COLLECTION")
+    } yield {
+      dataset.findValue("SF_DOMAIN")
```

Comment:
what is so special about the domain that it is not retrieved before the yield? I vagualy remember special treatment elsewhere.